iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/base_relative_pointer.hpp

```diff
@@ -51,28 +52,38 @@ namespace rp
 class BaseRelativePointer
 {
   public:
-    using id_t = uint64_t;
+    struct id_t : public cxx::NewType<uint64_t,
+                                      cxx::newtype::DefaultConstructable,
+                                      cxx::newtype::CopyConstructable,
+                                      cxx::newtype::Convertable,
+                                      cxx::newtype::ConstructByValueCopy,
+                                      cxx::newtype::MoveConstructable>
+    {
+        using ThisType::ThisType;
+    };
+
+    using id_underlying_t = id_t::value_type;
```

Comment:
fairly verbose.

A potential solution could be to call the struct `Id` (it is in the namespace so it is not ambiguous) and the underlying type `id_t`. This is consistent with the rule too only use  `_t` to aliases (which currently is violated). I personally only use `_t` for aliases to primitive types.

Reply:
`id_t` clashes with some system header definition and it not recommended to be used (see #1549). The current naming clearly states the intention, I'll leave it as it is.

Reply:
would not clash here due to namespace / in class definition though